02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,53 @@
+const { people } = require("../data");
+
+const getPeople = (req, res) => {
+  res.json(people);
+};
+
+const addPeople = (req, res) => {
+  if (!req.body.name) {
+    return res
+      .status(400)
+      .json({ success: false, message: "Please provide a name" });
+  }
+
+  const newPerson = {
+    id: people.length + 1,
+    name: req.body.name,
+  };
+
+  people.push(newPerson);
+
+  res.status(201).json({ success: true, name: req.body.name });
+};
+
+const getPerson = (req, res) => {
+  const id = parseInt(req.params.id);
+  const person = people.find((p) => p.id === id);
+
+  if (!person) {
+    return res
+      .status(404)
+      .json({ success: false, message: "Person not found" });
+  }
+
+  res.json(person);
+};
+
+const deletePerson = (req, res) => {
+  const id = parseInt(req.params.id);
+  const index = people.findIndex((p) => p.id === id);
+
+  if (index === -1) {
+    return res
+      .status(404)
+      .json({ success: false, message: "Person not found" });
+  }
+
+  const updatedPeople = people.filter((person)=> {
+    return parseInt(person.id) !== id
```

Comment:
Although the filter method is implemented correctly here, you are not removing the people from the people array. If we call the get endpoint, we can see that the people are still there.
To fix this, you can call the method splice to delete the people from the people array passing the index of person in the people array.
Here is an example of how you can do that:
people.splice(index, 1);

Reply:
Thank you. I already changed the code. Please check it again